src/main/java/org/redhat/sbomer/rest/v1alpha1/SBOMResource.java

```diff
@@ -116,50 +99,77 @@ public Page<Sbom> list(@Valid @BeanParam PaginationParameters paginationParams)
+        return sbomService
+                .listAllSbomsWithBuildId(buildId, paginationParams.getPageIndex(), paginationParams.getPageSize());
+    }
+
+    @POST
+    @Operation(
+            summary = "Generate a base SBOM based on the PNC build using Domino.",
+            description = "SBOM base generation for a particular PNC build Id offloaded to the service, using Domino.")
+    @Parameter(name = "buildId", description = "PNC build identifier", example = "ARYT3LBXDVYAC")
+    @Path("/generate/domino/{buildId}")
+    @APIResponses({ @APIResponse(
+            responseCode = "201",
+            description = "Schedules generation of a SBOM for a particular PNC buildId. This is an asynchronous call. It does execute the generation behind the scenes.",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON)) })
+    public Response generateFromBuildWithDomino(@PathParam("buildId") String id) throws Exception {
+
+        return sbomService.generateSbomFromPncBuild(id, Generators.DOMINO);
     }
 
     @POST
     @Operation(
-            summary = "Enrich SBOM based on the PNC build",
-            description = "SBOM enrichment for a particular PNC build Id. The only mode currently available is `ENRICHED_v1_0`")
+            summary = "Generate a base SBOM based on the PNC build using CycloneDX plugin.",
+            description = "SBOM base generation for a particular PNC build Id offloaded to the service, using CycloneDX plugin.")
     @Parameter(name = "buildId", description = "PNC build identifier", example = "ARYT3LBXDVYAC")
-    @Parameter(
-            name = "mode",
-            description = "Enrichment generation mode. Available modes: `ENRICHED_v1_0`",
-            example = "ENRICHED_v1_0")
-    @Path("/enrich/{buildId}")
+    @Path("/generate/cyclonedx/{buildId}")
     @APIResponses({ @APIResponse(
             responseCode = "201",
-            description = "Executes the enrichment of an existing SBOM for a particular PNC buildId.",
+            description = "Schedules generation of a SBOM for a particular PNC buildId. This is an asynchronous call. It does execute the generation behind the scenes.",
             content = @Content(mediaType = MediaType.APPLICATION_JSON)) })
-    public Response runEnrichmentOfBaseSbom(@PathParam("buildId") String buildId, @QueryParam("mode") String mode)
-            throws Exception {
+    public Response generateFromBuildWithCycloneDX(@PathParam("buildId") String id) throws Exception {
+
+        return sbomService.generateSbomFromPncBuild(id, Generators.CYCLONEDX);
+    }
 
+    @POST
+    @Operation(
+            summary = "Save the base SBOM and run and enrichment with PncToSbomProperties processor.",
+            description = "Save the base SBOM and run and enrichment with PncToSbomProperties processor.")
+    @Parameter(name = "sbom", description = "The SBOM to save")
+    @Path("/process/sbomproperties")
```

Comment:
Similar comment to previous one. If we would make it just `/process`  and move to params we could simplify the API and make it more less coupled.